tests/src/quickgui/app/main.cpp

```diff
@@ -0,0 +1,97 @@
+
+#include <QCommandLineParser>
+#include <QQmlApplicationEngine>
+#include <QQmlComponent>
+#include <QQmlContext>
+#include <QQmlError>
+
+#include "qgis.h"
+#include "qgsapplication.h"
+#include "qgsproject.h"
+#include "qgslayertree.h"
+#include "qgsmessagelog.h"
+#include "qgsquickutils.h"
+#include "qgslogger.h"
+
+int main( int argc, char *argv[] )
+{
+  // 1) Initialize QGIS
+  QgsApplication app( argc, argv, true );
+
+  // Set up the QSettings environment must be done after qapp is created
+  QCoreApplication::setOrganizationName( "QGIS" );
+  QCoreApplication::setOrganizationDomain( "qgis.org" );
+  QCoreApplication::setApplicationName( "QgsQuick Test App" );
+  QCoreApplication::setApplicationVersion( Qgis::QGIS_VERSION );
+
+  QCommandLineParser parser;
+  parser.addVersionOption();
+  parser.process( app );
+
+  QgsApplication::init();
+  QgsApplication::initQgis();
+
+  // 2) Load QGIS Project
+  QString dataDir( TEST_DATA_DIR );  // defined in CMakeLists.txt
+  QString projectFile = dataDir + "/quickapp_project.qgs";
+  QgsDebugMsg( QStringLiteral( "project file:  %1" ).arg( projectFile ) );
+  QgsProject project;
+  bool res = project.read( projectFile );
+  Q_ASSERT( res );
+
+  QQmlEngine engine;
+  engine.addImportPath( QgsApplication::qmlImportPath() );
+  engine.rootContext()->setContextProperty( "__project", &project );
+  engine.rootContext()->setContextProperty( "__layers", QVariant::fromValue( project.layerTreeRoot()->layerOrder() ) );
+
+  // Set simulated position for desktop builds
+  bool use_simulated_position = true;
+  engine.rootContext()->setContextProperty( "__use_simulated_position", use_simulated_position );
+
+  QQmlComponent component( &engine, QUrl( QStringLiteral( "qrc:/main.qml" ) ) );
+  QObject *object = component.create();
+
+  if ( !component.errors().isEmpty() )
+  {
+    QgsDebugMsg( QStringLiteral( "%s" ).arg( QgsApplication::showSettings().toLocal8Bit().data() ) );
+
+    QgsDebugMsg( QStringLiteral( "****************************************" ) );
+    QgsDebugMsg( QStringLiteral( "*****        QML errors:           *****" ) );
+    QgsDebugMsg( QStringLiteral( "****************************************" ) );
+    const QList<QQmlError> errors = component.errors();
+    for ( const QQmlError &error : errors )
+    {
+      QgsDebugMsg( error.toString() );
+    }
+    QgsDebugMsg( QStringLiteral( "****************************************" ) );
+    QgsDebugMsg( QStringLiteral( "****************************************" ) );
+  }
+
+  if ( object == nullptr )
+  {
+    QgsDebugMsg( QStringLiteral( "FATAL ERROR: unable to create main.qml" ) );
+    return EXIT_FAILURE;
+  }
+
+  // Add some data for debugging if needed
+  QgsApplication::messageLog()->logMessage( QgsQuickUtils::instance()->dumpScreenInfo() );
+  QgsApplication::messageLog()->logMessage( "data directory: " + dataDir );
+  QgsApplication::messageLog()->logMessage( "All up and running" );
```

Comment:
I think what ends up in the message log should be translated.

Reply:
does it apply to tests too?

Reply:
I don't think so, but there you'd probably use QgsDebugMsg (no reason to send it through the message log).

Reply:
Since this is kind-of test (in test/ directory), I do not know how it works but I would not like to add translators more work for some random string in test cases.

Reply:
ah, you are right. Forget it.
Can you have a look at the dump method in the QgsQuickUtils then?

Reply:
I am sorry, which method do you mean? If dumpScreenInfo, what is wrong with it?

Reply:
I think it doesn't live in test and I think it is untranslated, no?

Reply:
translated